File: src/main/api/utils/generateSiteId.ts
export const generateSiteId = (model: string = 'anime'): string => {
  let id: string

  switch (model.toLowerCase()) {
    case 'anime':
      id = '5'
      break
    case 'manga':
      id = '1'
      break
    case 'ranobe':
      id = '3'
      break
    case 'slash':
      id = '2'
      break
    case 'hentai':
      id = '4'
      break
    default:
      id = '5'
  }

  return id
}

File: src-tauri/src/api/utils/generate_site_id.rs
pub fn generate_site_id(model: &str) -> String {
    let model = model.to_lowercase();

    let id = match model.as_str() {
        "anime" => "5",
        "manga" => "1",
        "ranobe" => "3",
        "slash" => "2",
        "hentai" => "4",
        _ => "5",
    };

    id.to_string()
}
